types/validator_set.go

```diff
@@ -48,12 +48,17 @@ func NewValidatorSet(vals []*Validator) *ValidatorSet {
 }
 
 // incrementAccum and update the proposer
-// TODO: mind the overflow when times and votingPower shares too large.
 func (valSet *ValidatorSet) IncrementAccum(times int) {
 	// Add VotingPower * times to each validator and order into heap.
 	validatorsHeap := cmn.NewHeap()
 	for _, val := range valSet.Validators {
-		val.Accum += val.VotingPower * int64(times) // TODO: mind overflow
+		res, overflow := signedMulWithOverflowCheck(val.VotingPower, int64(times))
```

Comment:
s/signedMulWithOverflowCheck/mulSafe/g
I don't think we'd check overflow with unsigned numbers.

Reply:
or multiply unsigned numbers for that matter.  if we do it's probably for bit-twiddling and we'll prob want to create custom functions for those kinds of operations, I think.